javascript/javascript1/week3/Js1week3.js

```diff
@@ -0,0 +1,189 @@
+
+const notes = [];
+function saveNote(content, id) {
+    return notes.push({ content, id });
+
+}
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("water the plants", 3);
+
+
+
+function getNote(id) {
+    if (typeof id !== 'number' || typeof id === null || id > notes.length) {
+        return console.log(`It is an error string`);
+    }
+
+    else {
+
+        for (let i = 0; i <= notes.length; i++) {
+
+            if (id == notes[i].id) {
+                return console.log(notes[i])
+            }
+
+        }
+
+    }
+}
+
+getNote(3)
+//
+function logOutNotesFormatted() {
+    for (let i = 0; i < notes.length; i++) {
+
+        noteFormat = console.log(`The note with id: ${notes[i].id}, has the following note text: ${notes[i].content}.`)
+    }
+    return noteFormat;
+}
+logOutNotesFormatted()
+
+//  Usage limit
+let activities = [];
+function addActivity(date, activity, duration) {
+    if (typeof date === 'string' && typeof activity === 'string' && typeof duration === 'number')
+
+        return activities.push({ date, activity, duration })
+    else
+        return console.log(`It is not a valid data type.`)
+}
+
+
+addActivity("07/6-22", "Youtube", 30);
+addActivity("07/6-22", "TikTok", 30);
+addActivity()
+
+//addActivity("07/06-22", "Twitter", 60);
+
+function showStatus(activities) {
+    let activitySum = 0;
+    for (let i = 0; i < activities.length; i++) {
+        activitySum += activities[i].duration;
+    }
+    if (activities.length == 0) {
+        return console.log(`Add some activities before calling showStatus.`);
+    }
+
+    else if (activitySum <= 120) {
+
+        return console.log(`You have added ${activities.length} activities. They amount to ${activitySum} min. of usage`)
+    }
+    else
+        return console.log(`You have reached your limit, no more smartphoning for you!`)
+
+}
+showStatus(activities)
+
+
+// Extra feature
+// Twe don't need to specify the date,the function automatically figures out the date.
+
+const options = { weekday: 'narrow', year: '2-digit', month: '2-digit', day: 'numeric' };
+let today = new Date();
+date = today.toLocaleDateString(undefined, options);
+function addActivity(date, activity, duration) {
+    if (typeof date == 'string' && typeof activity == 'string' && typeof duration == 'number')
+        return activities.push({ date, activity, duration })
+}
+
+
+// Create a function for calculating the activity a user has spent the most time on.
+addActivity(date, "Youtube", 30);
+addActivity(date, "TikTok", 45);
+addActivity(date, "Twitter", 60);
+
+function activitySpentMore(activities) {
+    let mostSpend = activities[0].duration;
+    for (let i = 1; i < activities.length; i++) {
+        if (mostSpend < activities[i].duration) {
+            mostSpend = activities[i].duration;
+            mostSpendActivity = activities[i].activity;
+        }
+
+    }
+
+
+    return console.log(`You spend ${mostSpend} min on ${mostSpendActivity} `)
```

Comment:
Once again, you can omit the return keyword from this line 😊

Great work with the homework overall - you have a really strong sense of logic and some cool implementations 💫

Reply:
@MLMeyer Thank you so much for your feedback. I really appreciate that you take the time to review my code.